src/Access/Portal.vala

```diff
@@ -100,7 +100,7 @@ public class Access.Portal : Object {
             access_dialog.callback ();
         });
 
-        dialog.show_all ();
+        dialog.show ();
         yield;
 
```

Comment:
Use `present()` instead of `show()`, unregister and destroy the dialog after the yield.

```suggestion
        dialog.present ();
        yield;

        connection.unregister_object (dialog.register_id);
        dialog.destroy ();
```
the unrealize callback isn't needed anymore with this suggestion.